gateway/tests/api/authentication/custom_token/test_ibm_quantum_platform.py

```diff
@@ -104,3 +107,78 @@ def test_default_authentication_workflow(
             for group in user.groups.all():
                 permissions = list(group.permissions.values_list("codename", flat=True))
                 self.assertEqual(permissions, [VIEW_PROGRAM_PERMISSION])
+
+    @patch.object(IamAccessGroupsV2, "list_access_groups")
+    @patch.object(ResourceControllerV2, "get_resource_instance")
+    @patch.object(IamIdentityV1, "get_api_keys_details")
+    @responses.activate
+    def test_default_authentication_workflow_with_inactive_account(
+        self,
+        mock_get_api_keys_details: MagicMock,
+        mock_get_resource_instance: MagicMock,
+        mock_list_access_groups: MagicMock,
+    ):
+        """
+        This test verifies the entire flow of the custom token authentication
+        for a not verified user.
+        """
+
+        mock_get_api_keys_details.return_value = DetailedResponse(
+            response={
+                "iam_id": "IBMid-1000000XYZ",
+                "account_id": "abc18abcd41546508b35dfe0627109c4",
+            },
+            headers={},
+            status_code=200,
+        )
+
+        mock_get_resource_instance.return_value = DetailedResponse(
+            response={
+                "resource_plan_id": "f04b2f00-35b0-46b0-b84d-eb63418417e6",
+            },
+            headers={},
+            status_code=200,
+        )
+
+        mock_list_access_groups.return_value = DetailedResponse(
+            response={
+                "groups": [
+                    {
+                        "id": "AccessGroupId-23afbcd24-00a0-00ab-ab0c-1a23b4c567de",
+                        "name": "Private Group",
+                    }
+                ],
+            },
+            headers={},
+            status_code=200,
+        )
+
+        responses.add(
+            responses.POST,
+            "https://iam.test.cloud.ibm.com/identity/token",
+            json={
+                "iam_id": "IBMid-1000000XYZ",
+                "account_id": "abc18abcd41546508b35dfe0627109c4",
+            },
+            status=200,
+        )
+
+        custom_auth = CustomTokenBackend()
+        request = MagicMock()
+        request.META = {
+            "HTTP_SERVICE_CHANNEL": "ibm_quantum_platform",
+            "HTTP_AUTHORIZATION": "Bearer AWESOME_TOKEN",
```

Comment:
I am always unsure of when the bearer token is used vs the API key (which we call token in the runtime context).

Reply:
We can discuss it in detail tomorrow but basically the Bearer is an HTTP standard for authentication headers. You usually build your header with the format: `Bearer <token>` from a HTTP perspective. [MDN reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Authentication).

The Bearer schema it's not the only one we can follow for the authentication, another common ones are the Basic and Digest, for example. But I would say Bearer is the most common.